examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/main.c

```diff
@@ -130,469 +138,565 @@ static xi_state_t init_xively_topics( xi_context_handle_t in_context_handle );
+
+    if ( io_read_gyro( &sensor_input ) < 0 )
+    {
+        printf( "\r\n\t[ERROR] trying to read gyroscope input" );
+        return -1;
+    }
+    if ( io_axes_to_json( sensor_input, out_msg, IO_AXES_JSON_BUFFER_MAX_SIZE ) < 0 )
+    {
+        printf( "\r\n\t[ERROR] trying to create JSON string from sensor input" );
+        return -1;
+    }
+    return xi_publish( gXivelyContextHandle, mqtt_topic_descr->name, out_msg,
+                       XI_MQTT_QOS_AT_MOST_ONCE, XI_MQTT_RETAIN_FALSE, NULL, NULL );
 }
 
-xi_state_t pub_gyroscope( const mqtt_topic_descr_t* const mqtt_topic_descr )
+xi_state_t pub_barometer( const mqtt_topic_descr_t* const mqtt_topic_descr )
 {
-	return xi_publish( gXivelyContextHandle,
-			mqtt_topic_descr->name, "greetings from gyroscope",
-			XI_MQTT_QOS_AT_MOST_ONCE,
-			XI_MQTT_RETAIN_FALSE, NULL, NULL );
+    char out_msg[IO_FLOAT_BUFFER_MAX_SIZE] = "";
+    float sensor_input                     = 0;
+    io_read_pressure( &sensor_input );
+    io_float_to_string( sensor_input, out_msg, IO_FLOAT_BUFFER_MAX_SIZE );
+    return xi_publish( gXivelyContextHandle, mqtt_topic_descr->name, out_msg,
+                       XI_MQTT_QOS_AT_MOST_ONCE, XI_MQTT_RETAIN_FALSE, NULL, NULL );
 }
 
 xi_state_t pub_humidity( const mqtt_topic_descr_t* const mqtt_topic_descr )
 {
-	return xi_publish( gXivelyContextHandle,
-			mqtt_topic_descr->name, "greetings from humidity",
-			XI_MQTT_QOS_AT_MOST_ONCE,
-			XI_MQTT_RETAIN_FALSE, NULL, NULL );
+    char out_msg[IO_FLOAT_BUFFER_MAX_SIZE] = "";
+    float sensor_input                     = 0;
+    io_read_humidity( &sensor_input );
+    io_float_to_string( sensor_input, out_msg, IO_FLOAT_BUFFER_MAX_SIZE );
+    return xi_publish( gXivelyContextHandle, mqtt_topic_descr->name, out_msg,
+                       XI_MQTT_QOS_AT_MOST_ONCE, XI_MQTT_RETAIN_FALSE, NULL, NULL );
 }
 
 xi_state_t pub_temperature( const mqtt_topic_descr_t* const mqtt_topic_descr )
 {
-	return xi_publish( gXivelyContextHandle,
-			mqtt_topic_descr->name, "greetings from temperature",
-			XI_MQTT_QOS_AT_MOST_ONCE,
-			XI_MQTT_RETAIN_FALSE, NULL, NULL );
+    char out_msg[IO_FLOAT_BUFFER_MAX_SIZE] = "";
+    float sensor_input                     = 0;
+    io_read_temperature( &sensor_input );
+    io_float_to_string( sensor_input, out_msg, IO_FLOAT_BUFFER_MAX_SIZE );
+    return xi_publish( gXivelyContextHandle, mqtt_topic_descr->name, out_msg,
+                       XI_MQTT_QOS_AT_MOST_ONCE, XI_MQTT_RETAIN_FALSE, NULL, NULL );
+}
+
+xi_state_t pub_button( const mqtt_topic_descr_t* const mqtt_topic_descr )
+{
+    char* payload;
+    ( io_read_button() ) ? ( payload = "1" ) : ( payload = "0" );
+    return xi_publish( gXivelyContextHandle, mqtt_topic_descr->name, payload,
+                       XI_MQTT_QOS_AT_MOST_ONCE, XI_MQTT_RETAIN_FALSE, NULL, NULL );
+}
+
+void pub_button_interrupt( void )
+{
+    /* TODO: Verify we are connected to Xively before calling xi_publish */
+    char payload[] = "1";

```

Comment:
:+1:

Regarding that TODO; is there a recommended way to query libxively for its connection status? Or do I need to keep track of it at the application level?